test/new-e2e/cws/e2e_cws_test.go

```diff
@@ -0,0 +1,150 @@
+	e2e.Run(t, &agentSuite{}, e2e.AgentStackDef(
+		[]ec2params.Option{
+			ec2params.WithName("cws-e2e-tests"),
+		},
+		agentparams.WithAgentConfig(agentConfig),
+		agentparams.WithSecurityAgentConfig("runtime_security_config:\n  enabled: true"),
+		agentparams.WithSystemProbeConfig("system_probe_config:\n  enabled: true\n  log_level: trace\nruntime_security_config:\n  enabled: true\nlog_patterns:\n  - module.APIServer.*"),
+		agentparams.WithVersion("7.46.0"),
+	), params.WithDevMode())
+}
+
+func (a *agentSuite) SetupSuite() {
+	// Create temporary directory
+	tempDir := a.Env().VM.Execute("mktemp -d")
+	a.dirname = strings.TrimSuffix(tempDir, "\n")
+	a.filename = fmt.Sprintf("%s/secret", a.dirname)
+	a.testID = uuid.NewString()[:4]
+	a.desc = fmt.Sprintf("e2e test rule %s", a.testID)
+	a.agentRuleName = fmt.Sprintf("e2e_agent_rule_%s", a.testID)
+	a.Suite.SetupSuite()
+}
+
+func (a *agentSuite) TearDownSuite() {
+
+	if len(a.signalRuleID) != 0 {
+		a.apiClient.DeleteSignalRule(a.signalRuleID)
+	}
+	if len(a.agentRuleID) != 0 {
+		a.apiClient.DeleteAgentRule(a.agentRuleID)
+	}
+	a.Env().VM.Execute(fmt.Sprintf("rm -r %s", a.dirname))
+	a.Suite.TearDownSuite()
+}
+
+func (a *agentSuite) TestOpenSignal() {
+	a.apiClient = cws.NewAPIClient()
+
+	// Create CWS Agent rule
+	rule := fmt.Sprintf("open.file.path == \"%s\"", a.filename)
+	res, err := a.apiClient.CreateCWSAgentRule(a.agentRuleName, a.desc, rule)
+	require.NoError(a.T(), err, "Agent rule creation failed")
+	a.agentRuleID = res.Data.GetId()
+
+	// Create Signal Rule (backend)
+	res2, err := a.apiClient.CreateCwsSignalRule(a.desc, "signal rule for e2e testing", a.agentRuleName, []string{})
+	require.NoError(a.T(), err, "Signal rule creation failed")
+	a.signalRuleID = res2.GetId()
+
+	// Check if the agent has started
+	isReady := a.Env().Agent.IsReady()
+	assert.Equal(a.T(), isReady, true, "Agent should be ready")
+
+	// Check if the agent use the right configuration
+	assert.Contains(a.T(), a.Env().Agent.Config(), "log_level: DEBUG")
+
+	// Check if system-probe has started
+	err = cws.WaitAgentLogs(a.Env().VM, "system-probe", cws.SystemProbeStartLog)
+	require.NoError(a.T(), err, "system-probe could not start")
+
+	// Check if security-agent has started
+	err = cws.WaitAgentLogs(a.Env().VM, "security-agent", cws.SecurityStartLog)
+	require.NoError(a.T(), err, "security-agent could not start")
+
+	// Download policies
+	apiKey, err := runner.GetProfile().SecretStore().Get(parameters.APIKey)
+	require.NoError(a.T(), err, "Could not get API KEY")
+
+	appKey, err := runner.GetProfile().SecretStore().Get(parameters.APPKey)
+	require.NoError(a.T(), err, "Could not get APP KEY")
+
+	policies := a.Env().VM.Execute(fmt.Sprintf("DD_APP_KEY=%s DD_API_KEY=%s DD_SITE=datadoghq.com %s runtime policy download", appKey, apiKey, cws.SecurityAgentPath))
+
+	assert.NotEmpty(a.T(), policies, "should not be empty")
+	a.policies = policies
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
You might need to retry this part too, in case of network issues for example. You can use `agentSuite.EventuallyWithT` and pass a function with the assert. It will retry until timeout or until assert is valid